tools/gromacs/rmsd.xml

```diff
@@ -0,0 +1,150 @@
+<tool id="gmx_rmsd" name="GROMACS RMSD calculation" version="@TOOL_VERSION@+galaxy@GALAXY_VERSION@">
+    <description>of molecular structures</description>
+    <macros>
+        <import>macros.xml</import>
+        <token name="@GALAXY_VERSION@">0</token>
+    </macros>
+    <expand macro="requirements"/>
+    <command detect_errors="exit_code"><![CDATA[
+
+    ln -s '$traj_input' ./traj_input.${traj_input.ext} &&
+    ln -s '$structure_input' ./structure_input.${structure_input.ext} &&
+    #if '$traj2.traj2select' == 'Yes':
+    ln -s '$traj2.traj_input2' ./traj_input2.${traj_input2.ext} &&
+    #end if
+    #if '$ndx_input':
+    ln -s '$ndx_input' ./ndx_input.${ndx_input.ext} &&
+    #end if
+    #if '$fit.fit_type' == 'none':
+    echo '$index_rmsdgroup' | gmx rms
+    #else
+    echo '$fit.index_refgroup' '$index_rmsdgroup' | gmx rms
+    #end if
+    ##inputs
+    -f ./traj_input.${traj_input.ext}  -s ./structure_input.${structure_input.ext}
+    #if '$traj2.traj2select' == 'Yes':
+    -f2 ./traj_input2.${traj_input2.ext}
+    #end if
+    #if '$ndx_input':
+    -n ./ndx_input.${ndx_input.ext}
+    #end if
+
+    -fit '$fit.fit_type'
+    -xvg '$outputformat'
+    ## other options
+    #if $mir == 'true':
+    -mir
+    #end if
+
+    >> verbose.txt 2>&1
+ 
+    ]]></command>
+    <inputs>
+        <param name="traj_input" type="data" format="trr,xtc" label="Trajectory file" help="In XTC or TRR format"/>
+        <conditional name="traj2">
+            <param name="traj2select" type="select" label="Add an additional trajectory file?" help="Can be useful when compraring two different simulations of the same system.">
+                <option value="No">No</option>
+                <option value="Yes">Yes</option>
+            </param>
+            <when value="Yes">
+                <param name="traj_input2" type="data" format="trr,xtc" label="Additional trajectory file" help="In XTC or TRR format."/>
```

Comment:
I would remove the `.` ad the and also in the select options below

Reply:
Why do you need this conditional here, can you not make the data-param optional=true

Reply:
So the problem I'm encountering by doing this, is that this always assigns a value to $traj_input2. Even if nothing is selected, it assigns the value of 'None' to this parameter. So because of this, when I'm running the tests, particularly the test that doesn't involve a second trajectory, the command line still executes '...-f2 traj_input2...', which causes the tool to fail planemo test, since gromacs receives 'traj_input2' as just a blank .dat file.

Reply:
You can use something like
```
#if $traj_input2:
    -f2 $traj_input2
#end if
```
in the command

Reply:
Yeah that's one of the things I was trying and it still wasn't working. the 'optional=true'  flag means that there is always a selection, even if its it nothing, and when nothing is selected, '-f2 None.dat' goes into the command line. the conditional force there to actually be. I even tried:

#if '$traj_input2' != 'None':
      -f2 '$traj_input2'
#end if 

but that still doesn't work.

Reply:
Try `#if str($inputs.cov) == 'None':`

(it can help to search the existing tools on the repo or tools-iuc for solutions, I found the above in https://github.com/galaxycomputationalchemistry/galaxy-tools-compchem/blob/master/tools/biomoldyn/fastpca.xml#L13)